sdk/storage/azure-storage-files-datalake/src/datalake_path_client.cpp

```diff
@@ -271,14 +283,29 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
       const DeletePathOptions& options,
       const Azure::Core::Context& context) const
   {
-    _detail::PathClient::DeletePathOptions protocolLayerOptions;
-    protocolLayerOptions.LeaseId = options.AccessConditions.LeaseId;
-    protocolLayerOptions.IfMatch = options.AccessConditions.IfMatch;
-    protocolLayerOptions.IfNoneMatch = options.AccessConditions.IfNoneMatch;
-    protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
-    protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
-    protocolLayerOptions.Recursive = options.Recursive;
-    return _detail::PathClient::Delete(*m_pipeline, m_pathUrl, protocolLayerOptions, context);
+    bool paginated = m_clientConfiguration.ApiVersion >= "2023-08-03"
+        && m_clientConfiguration.TokenCredential != nullptr;
+    std::string continuationToken = "";
+    do
```

Comment:
It's considered a bad practice to use `do while` loops in some style guides. Some formatter would emit a warning.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-do